Visitor.NET.AutoVisitableGen/Visitor.NET.AutoVisitableGen.Tests/AutoVisitableIncrementalSourceGeneratorTests.cs

```diff
@@ -71,6 +71,72 @@ namespace MyNamespace;
             ignoreLineEndingDifferences: true);
     }
 
+    [Theory]
+    [InlineData("class")]
+    [InlineData("record")]
+    public void CorrectlyGeneratedSourceTest2(string keyword)
```

Comment:
need to rename test and the one above. Let's stick to the Roy Osherove convention

My suggestions are:

- `Initialize_DerivedTypeVisitable_ImplementationCorrectlyGenerated`
- `Initialize_NestedDerivedTypeVisitable_NestedImplementationCorrectlyGenerated`

Reply:
Need more tests to see how the feature works with greater levels of depth.
i.e.

```cs
public partial class MyClassLevelOne
{
    public partial class MyClassLevelTwo
    {
        [AutoVisitable<BinaryTreeNode>]
        public partial record Number(double Value) : BinaryTreeNode;
    }
}
```

And so on with increasing depth levels

Reply:
using your suggestions to tests names

Reply:
What do you mean by increasing depth levels? is your example sufficient or do you mean to geneate like 'dynamic' levels based on test case number ?

Reply:
my example is sufficient

Reply:
not sure i understand what you are saying, can you give an example ?

Reply:
https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/

your data will consist of three string pairs: input and expected output.
inputs are:
```cs
public partial class MyClassLevelOne
{
    public partial class MyClassLevelTwo
    {
        [AutoVisitable<BinaryTreeNode>]
        public partial record Number(double Value) : BinaryTreeNode;
    }
}
```
```cs
public partial class MyClassLevelOne
{
    public partial class MyClassLevelTwo
    {
        public partial class MyClassLevelThree
        {
            [AutoVisitable<BinaryTreeNode>]
            public partial record Number(double Value) : BinaryTreeNode;
        }
    }
}
```
```cs
public partial class MyClassLevelOne
{
    public partial class MyClassLevelTwo
    {
        public partial class MyClassLevelThree
        {
            public partial class MyClassLevelFour
            {
                [AutoVisitable<BinaryTreeNode>]
                public partial record Number(double Value) : BinaryTreeNode;
            }
        }
    }
}
```

Reply:
Added tests, I placed tests data in another class like shown [here](https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/#loading-data-from-a-property-or-method-on-a-different-class)
because it took alot of space in the tests file